File: 0414-third-maximum-number/0414-third-maximum-number.java
class Solution {
    public int thirdMax(int[] nums) {
        long max1 = Long.MIN_VALUE;
        Long max2 = Long.MIN_VALUE;
        long max3 = Long.MIN_VALUE;

        for(long x : nums){

            if(x == max1 || x == max2 || x == max3){
                continue;
            }

            if(x > max1){
                max3 = max2;
                max2 = max1;
                max1 = x;
            }
            else if(x > max2){
                max3 = max2;
                max2 = x;
            }
            else if(x > max3){
                max3 = x;
            }
        }
        if(max3 == Long.MIN_VALUE){
            return (int) max1;
        }
        return (int)max3;
    }
}

File: 0414-third-maximum-number/0414-third-maximum-number.py
class Solution(object):
    def thirdMax(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        max2 = float('-inf')
        max1 = float('-inf')
        max3 = float('-inf')

        for i in nums:
            
            if(i == max1 or i == max2 or i == max3):
                continue

            if(i > max1):
                max3 = max2
                max2 = max1
                max1 = i
            elif(i > max2):
                max3 = max2
                max2 = i
            elif(i > max3):
                max3 = i
        
        if(max3 == float('-inf')):
            return max1
        return max3